src/tool_cb_hdr.c

```diff
@@ -103,14 +116,26 @@ size_t tool_header_cb(char *ptr, size_t size, size_t nmemb, void *userdata)
   if(per->config->etag_save_file && etag_save->stream) {
     /* match only header that start with etag (case insensitive) */
     if(curl_strnequal(str, "etag:", 5)) {
-      char *etag_start = ptr + strlen("ETag:");
+      const char *etag_start = ptr + strlen("ETag:");
```

Comment:
I think you could use '5' instead of a strlen() call here since 5 is used in the line above already. (Then we can ignore the odd fact that the two etag strings are specified using different casings.) Perhaps just:
~~~c
const char *etag_start = &ptr[5];

Reply:
See https://github.com/curl/curl/pull/5832